cookbook/controller/argument_value_resolver.rst

```diff
@@ -0,0 +1,163 @@
+    :class:``Symfony\\Component\\HttpKernel\\ControllerMetadata\\ArgumentMetadataFactory``. This
+    factory will work on every supported PHP version but might give different results. E.g. the
+    ``isVariadic()`` will never return true on PHP 5.5 and only on PHP 7.0 and higher it will give
+    you basic types when calling ``getType()``.
+
+Now that you know what to do, you can implement this interface. In order to get the current ``User``,
+you will have to get it from the ``TokenInterface`` which is in the ``TokenStorageInterface``::
+
+    namespace AppBundle\ArgumentValueResolver;
+
+    use AppBundle\User;
+    use Symfony\Component\HttpKernel\Controller\ArgumentValueResolverInterface;
+    use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+
+    class UserValueResolver implements ArgumentValueResolverInterface
+    {
+        private $tokenStorage;
+
+        public function __construct(TokenStorageInterface $tokenStorage)
+        {
+            $this->tokenStorage = $tokenStorage;
+        }
+
+        public function supports(Request $request, ArgumentMetadata $argument)
+        {
+            return ($token = $this->tokenStorage->getToken()) && $token->getUser() instanceof User;
+        }
+
+        public function resolve(Request $request, ArgumentMetadata $argument)
+        {
+            yield $this->tokenStorage->getToken()->getUser();
+        }
+    }
+
+This was pretty simple, now all you have to do is add the configuration for the service container. This
+can be done by tagging the service with ``kernel.argument_resolver`` and adding a priority.
+
+.. note::
+
+    While adding a priority is optional, it's recommended to add one to make sure the expected
+    value is injected. The ``ArgumentFromAttributeResolver`` has a priority of 100. As this
+    one is responsible for fetching attributes from the ``Request``, it's also recommended to
+    trigger your custom value resolver with a lower priority. This makes sure the argument
+    resolvers are not triggered in (e.g.) subrequests if you pass your user along:
+    ``{{ render(controller('AppBundle:User:index', {'user', app.user})) }}``.
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # app/config/services.yml
+        services:
+            app.value_resolver.user:
+                class: AppBundle\ArgumentValueResolver\UserValueResolver
+                arguments:
+                    - '@security.token_storage'
+                tags:
+                    - { name: kernel.argument_resolver, priority: 50 }
+
+    .. code-block:: xml
+
+        <!-- app/config/services.xml -->
+        <services>
+            <service id="app.value_resolver.user" class="AppBundle\ArgumentValueResolver\UserValueResolver">
+                <argument type="service" id="security.token_storage">
+                <tag name="kernel.argument_resolver" priority="50" />
+            </service>
+        </services>
```

Comment:
That one should be included in the 2.3 version of the docs and not master. So if you have some extra time, please open a new PR. Otherwise, I don't really care that much as we are missing them in a lot of places.